lib/elixir/lib/file/stream.ex

```diff
@@ -73,7 +73,8 @@ defmodule File.Stream do
       start_fun =
         fn ->
           case :file.open(path, read_modes(modes)) do
-            {:ok, device}    -> device
+            {:ok, device}    ->
+              if :strip_bom in modes, do: strip_bom(device), else: device
```

Comment:
I don't like this solution. You don't need to know the logic of whether to stripping BOM in a function whose purpose is to *strip* the BOM, IMO. This usage of guards is not rare in Elixir though :) But if you don't want to go with `when strip_bom?`, I think the `if` solution looks alright.